sunpy/physics/differential_rotation.py

```diff
@@ -84,138 +88,318 @@ def diff_rot(duration: u.s, latitude: u.deg, rot_type='howard', frame_time='side
 
     A, B, C = rot_params[rot_type]
 
+    # This calculation of the rotation assumes a sidereal frame time.
     rotation = (A + B * sin2l + C * sin4l) * duration
 
+    # Applying this correction assumes that the observer is on the Earth,
+    # and that the Earth is at the same distance from the Sun at all times
+    # during the year.
     if frame_time == 'synodic':
         rotation -= 0.9856 * u.deg / u.day * duration
 
     return Longitude(rotation.to(u.deg))
 
 
-def solar_rotate_coordinate(coordinate,
-                            new_observer_time,
-                            new_observer_location="earth",
-                            **diff_rot_kwargs):
+def _get_new_observer(initial_obstime, observer, time):
```

Comment:
```suggestion
def _get_new_observer(initial_obstime, observer=None, time=None):
```
Unless None are meant to be passed in?